src/components/tree-detail/tree-water-needs/contact-dialog.tsx

```diff
@@ -0,0 +1,216 @@
+/* eslint-disable max-lines */
+import React, { useCallback, useState } from "react";
+import { useI18nStore } from "../../../i18n/i18n-store";
+import { AlertDialog } from "../../alert-dialog/alert-dialog";
+import { PrimaryButton } from "../../buttons/primary";
+import { TertiaryButton } from "../../buttons/tertiary";
+import { CheckIcon } from "../../icons/check-icon";
+import { CloseIcon } from "../../icons/close-icon";
+import Markdown from "react-markdown";
+import { supabaseClient } from "../../../auth/supabase-client";
+
+const closeContactDialog = () => {
+	(document.getElementById("contact-dialog") as HTMLDialogElement).close();
+};
+
+interface ContactDialogProps {
+	contactUsername: string;
+}
+export const ContactDialog: React.FC<ContactDialogProps> = ({
+	contactUsername,
+}) => {
+	const [message, setMessage] = useState("");
+	const [isContactRequestLoading, setIsContactRequestLoading] = useState(false);
+	const [alreadySentContact, setAlreadySentContact] = useState(false);
+	const [error, setError] = useState("");
+	const [
+		alreadySentMoreThan3RequestsInLast24Hours,
+		setAlreadySentMoreThan3RequestsInLast24Hours,
+	] = useState(false);
+
+	const i18n = useI18nStore().i18n();
+
+	const showContactSuccessDialog = () => {
+		(
+			document.getElementById("contact-successful-alert") as HTMLDialogElement
+		).showModal();
+		setTimeout(() => {
+			(
+				document.getElementById("contact-successful-alert") as HTMLDialogElement
+			).close();
+		}, 2000);
+	};
+
+	const onSubmit = useCallback(
+		async (e: React.FormEvent<HTMLFormElement>) => {
+			setError("");
+			setAlreadySentContact(false);
+			setAlreadySentMoreThan3RequestsInLast24Hours(false);
+			setIsContactRequestLoading(true);
+
+			e.preventDefault();
```

Comment:
as I mentioned above, here you can get the message via something along the lines of:
```typescript
const message = e.currentTarget.message.value
```